[Bug] Integration tests fail due to bash arithmetic and leftover database files

## Description
- `make integration-test` command fails immediately after the first passing test due to two issues in the test runner script.
- `db.ez` test fails on second run. But success on first run.

## Environment
- OS: Linux (Ubuntu/WSL)
- Branch: SchoolyB/main

## Steps to Reproduce

```bash
git clone https://github.com/SchoolyB/EZ.git
cd EZ
make install
make integration-test

### Actual Result

```bash
sullung@DESKTOP-1MJVMNK:~/Development/EZ$ make integration-test 
Building EZ...
go build -ldflags "-X main.Version=9254fcd -X main.BuildTime=2026-01-10_10:41:38" -o ez ./cmd/ez
Build complete! Binary: ./ez
./integration-tests/run_tests.sh
========================================
  EZ Integration Test Suite
========================================

Running PASS tests...
----------------------------------------
  core/arrays... PASS
make: *** [Makefile:61: integration-test] Error 
```

---

```bash
sullung@DESKTOP-1MJVMNK:~/Development/EZ$ ./ez integration-tests/pass/stdlib/db.ez 
warning[W2009]: accessing member 'message' on error type which may be nil - consider checking for nil first
  --> integration-tests/pass/stdlib/db.ez:193:18
   |
193 |         println("Failed: ${save_err.message}")
   |                  ^ accessing member on potentially nil value
   |

summary: generated 1 warning
=== @db Standard Library Test ===
  -- Opening --
   [PASS] db.open(string)
  -- Setting --
   [PASS] db.set(database, string, string)
  -- Getting --
   [PASS] db.get(database, string)
   [PASS] db.get(database, string)
  -- db.has --
   [PASS] db.has(database, string)
   [PASS] db.has(database, string)
  -- db.keys --
   [PASS] db.keys(database)
  -- db.prefix --
   [PASS] db.prefix(database,string)
  -- db.count --
   [PASS] db.count(database)
  -- db.delete --
   [PASS] db.delete(database, string)
   [PASS] db.delete(database, string)
  -- db.save --
   [PASS] db.save(database)
  -- db.exists --
   [PASS] db.exists(string) returns false for missing file
   [PASS] db.exists(string) returns true for existing file
  -- db.update_key_name --
   [PASS] db.update_key_name() returns true
   [PASS] db.update_key_name() old key removed
   [PASS] db.update_key_name() new key has value
   [PASS] db.update_key_name() returns false for missing key
  -- db.sort --
   [PASS] db.sort(db.ALPHA)
   [PASS] db.sort(db.ALPHA_DESC)
   [PASS] db.sort(db.VALUE_ALPHA)
   [PASS] db.sort(db.KEY_LEN)
   [PASS] db.sort(db.VALUE_LEN)
   [PASS] db.sort(db.NUMERIC)
   [PASS] db.sort(db.NUMERIC_DESC)
  -- db.close --
   [PASS] db.close(database)

=== Summary ===
  Passed: 26
  Failed: 0

ALL TESTS PASSED
sullung@DESKTOP-1MJVMNK:~/Development/EZ$ ./ez integration-tests/pass/stdlib/db.ez 
warning[W2009]: accessing member 'message' on error type which may be nil - consider checking for nil first
  --> integration-tests/pass/stdlib/db.ez:193:18
   |
193 |         println("Failed: ${save_err.message}")
   |                  ^ accessing member on potentially nil value
   |

summary: generated 1 warning
=== @db Standard Library Test ===
  -- Opening --
   [PASS] db.open(string)
  -- Setting --
   [PASS] db.set(database, string, string)
  -- Getting --
   [PASS] db.get(database, string)
   [PASS] db.get(database, string)
  -- db.has --
   [PASS] db.has(database, string)
   [PASS] db.has(database, string)
  -- db.keys --
   [FAIL] db.keys(database): unknown key found in database
  -- db.prefix --
   [PASS] db.prefix(database,string)
  -- db.count --
   [PASS] db.count(database)
  -- db.delete --
   [PASS] db.delete(database, string)
   [PASS] db.delete(database, string)
  -- db.save --
   [PASS] db.save(database)
  -- db.exists --
   [PASS] db.exists(string) returns false for missing file
   [PASS] db.exists(string) returns true for existing file
  -- db.update_key_name --
   [PASS] db.update_key_name() returns true
   [PASS] db.update_key_name() old key removed
   [PASS] db.update_key_name() new key has value
   [PASS] db.update_key_name() returns false for missing key
  -- db.sort --
   [PASS] db.sort(db.ALPHA)
   [PASS] db.sort(db.ALPHA_DESC)
   [PASS] db.sort(db.VALUE_ALPHA)
   [PASS] db.sort(db.KEY_LEN)
   [PASS] db.sort(db.VALUE_LEN)
   [PASS] db.sort(db.NUMERIC)
   [PASS] db.sort(db.NUMERIC_DESC)
  -- db.close --
   [PASS] db.close(database)

=== Summary ===
  Passed: 25
  Failed: 1

SOME TESTS FAILED
```

## Root Cause

### Post-increment operation
In run_tests.sh, the script uses set -e (exit on error) combined with post-increment ((PASS_COUNT++)).

When PASS_COUNT=0, the expression ((PASS_COUNT++)), which bash treats as false (exit code 1), causing immediate script termination.

### Leftover .ezdb files
The db.ez test creates database files that persist between runs, causing subsequent test runs to fail with "unknown key found in database".

## Proposal
1. Change post-increment to pre-increment -> `((++PASS_COUNT))` and `((++FAIL_COUNT))`
2. Add cleanup at the start of the script like, `rm -f "$PROJECT_ROOT"/*.ezdb`

--- a/integration-tests/pass/stdlib/db.ez
+++ b/integration-tests/pass/stdlib/db.ez
@@ -2,7 +2,7 @@
  * db.ez - Test @db standard library with PROPER ASSERTIONS
  */
 
-import @std, @db, @json
+import @std, @db, @json, @io
 using std
 
 const User struct {
@@ -381,6 +381,14 @@ do main() {
         failed += 1
     }
 
+    // ==================== CLEANUP ====================
+    println("  -- Cleanup --")
+
+    // Clean up test database files created during tests
+    temp _, _ = io.remove("mydb.ezdb")
+    temp _, _ = io.remove("rename_test.ezdb")
+    temp _, _ = io.remove("sort_test.ezdb")
+    println("   [INFO] Cleaned up test database files")
 
     // ==================== Summary ====================
     println("")

--- a/integration-tests/run_tests.sh
+++ b/integration-tests/run_tests.sh
@@ -52,14 +52,14 @@ for test_file in "$SCRIPT_DIR"/pass/core/*.ez; do
         if output=$("$EZ_BIN" "$test_file" 2>&1); then
             if echo "$output" | grep -q "SOME TESTS FAILED"; then
                 echo -e "${RED}FAIL${NC} (test assertions failed)"
-                ((FAIL_COUNT++))
+                ((++FAIL_COUNT))
             else
                 echo -e "${GREEN}PASS${NC}"
-                ((PASS_COUNT++))
+                ((++PASS_COUNT))
             fi
         else
             echo -e "${RED}FAIL${NC} (execution error)"
-            ((FAIL_COUNT++))
+            ((++FAIL_COUNT))
         fi
     fi
 done
@@ -73,14 +73,14 @@ for test_file in "$SCRIPT_DIR"/pass/stdlib/*.ez; do
         if output=$("$EZ_BIN" "$test_file" 2>&1); then
             if echo "$output" | grep -q "SOME TESTS FAILED"; then
                 echo -e "${RED}FAIL${NC} (test assertions failed)"
-                ((FAIL_COUNT++))
+                ((++FAIL_COUNT))
             else
                 echo -e "${GREEN}PASS${NC}"
-                ((PASS_COUNT++))
+                ((++PASS_COUNT))
             fi
         else
             echo -e "${RED}FAIL${NC} (execution error)"
-            ((FAIL_COUNT++))
+            ((++FAIL_COUNT))
         fi
     fi
 done
@@ -97,10 +97,10 @@ for dir in "$SCRIPT_DIR"/pass/multi-file/*/; do
 
             if output=$("$EZ_BIN" "$main_file" 2>&1); then
                 echo -e "${GREEN}PASS${NC}"
-                ((PASS_COUNT++))
+                ((++PASS_COUNT))
             else
                 echo -e "${RED}FAIL${NC} (execution error)"
-                ((FAIL_COUNT++))
+                ((++FAIL_COUNT))
             fi
         fi
     fi
@@ -124,10 +124,10 @@ if [ -d "$SCRIPT_DIR/pass/warnings" ]; then
 
             if echo "$output" | grep -q "warning\[$expected_warning\]"; then
                 echo -e "${GREEN}PASS${NC}"
-                ((PASS_COUNT++))
+                ((++PASS_COUNT))
             else
                 echo -e "${RED}FAIL${NC} (expected warning $expected_warning not found)"
-                ((FAIL_COUNT++))
+                ((++FAIL_COUNT))
             fi
         fi
     done
@@ -145,10 +145,10 @@ for test_file in "$SCRIPT_DIR"/fail/errors/*.ez; do
 
         if "$EZ_BIN" "$test_file" >/dev/null 2>&1; then
             echo -e "${RED}FAIL${NC} (expected error, got success)"
-            ((FAIL_COUNT++))
+            ((++FAIL_COUNT))
         else
             echo -e "${GREEN}PASS${NC}"
-            ((PASS_COUNT++))
+            ((++PASS_COUNT))
         fi
     fi
 done
@@ -164,10 +164,10 @@ for test_file in "$SCRIPT_DIR"/fail/multi-file/*.ez; do
 
         if "$EZ_BIN" "$test_file" >/dev/null 2>&1; then
             echo -e "${RED}FAIL${NC} (expected error, got success)"
-            ((FAIL_COUNT++))
+            ((++FAIL_COUNT))
         else
             echo -e "${GREEN}PASS${NC}"
-            ((PASS_COUNT++))
+            ((++PASS_COUNT))
         fi
     fi
 done
@@ -183,10 +183,10 @@ for dir in "$SCRIPT_DIR"/fail/multi-file/*/; do
 
             if "$EZ_BIN" "$main_file" >/dev/null 2>&1; then
                 echo -e "${RED}FAIL${NC} (expected error, got success)"
-                ((FAIL_COUNT++))
+                ((++FAIL_COUNT))
             else
                 echo -e "${GREEN}PASS${NC}"
-                ((PASS_COUNT++))
+                ((++PASS_COUNT))
             fi
         fi
     fi

